### DISCUSSION TITLE
Sortable Column with Pandas Dataframe

### DISCUSSION THREAD

**Discussion by BlackF0rest** [zauberzeug/nicegui#2823] [Q&A] [answered]
### Question

Hi,
I am generating my `ui.table` with `ui.table.from_pandas(dataframe)` and need to make a column sortable. Is there any way to do that? In the doc i didnt find anything showing anything similar.

**Comment by falkoschindler** [comment]
Hi @BlackF0rest,

You can manipulate the column definitions after instantiating the AG Grid element like this:
```py
df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
aggrid = ui.aggrid.from_pandas(df)
aggrid.options['columnDefs'][0]['sortable'] = True
```

**Reply by BlackF0rest** [reply]
So there is no way to do it with a table Element? I find it much nicer and have allready build an Website around it.

**Reply by falkoschindler** [reply, marked as answer]
Oh, I confused `ui.table` with `ui.aggrid`. Here you go:

```py
df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
table = ui.table.from_pandas(df)
table.columns[0]['sortable'] = True
```
